subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/DryRunLoggingIntegrationTest.groovy

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.internal.logging
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.internal.SystemProperties
+
+class DryRunLoggingIntegrationTest extends AbstractIntegrationSpec {
+    private static final String EOL = SystemProperties.instance.lineSeparator
+
+    def setup() {
+        executer.withArgument('--dry-run')
+        buildFile << """
+            task foo {}
+            task bar(dependsOn: foo) {}
+            task baz(dependsOn: bar) {}
+"""
+    }
+
+    def "all tasks are printed when dry run mode is enabled using rich console"() {
+        given:
+        executer.withRichConsole()
+
+        when:
+        succeeds('baz')
+
+        then:
+        output =~ /(?s).*> Task :foo.*> Task :bar.*> Task :baz.*/
+    }
+
+    def "all tasks are printed when dry run mode is enabled using plain console"() {
+        given:
+        executer.withArgument('--console=plain')
+
+        when:
+        succeeds('baz')
+
+        then:
+        result.output.contains(":foo SKIPPED${EOL}:bar SKIPPED${EOL}:baz SKIPPED")
```

Comment:
couldn't this just use the built-in parsing we have in the result?

e.g., `result.assertTasksSkipped(":foo", ":bar", ":baz")`